core/src/com/sonyericsson/chkbugreport/util/Util.java

```diff
@@ -732,4 +748,53 @@ public static void printResource(Renderer r, String resName) {
         }
     }
 
+    /**
+     * Parse a UID as an integer.
+     *
+     * Handles both integer type and string type:
+     *   - 1000
+     *   - 1020
+     *   - u0a123
+     *
+     * Decoding from:
+     *   - libc/bionic/grp_pwd.cpp
+     *   - frameworks/base/core/java/android/os/UserHandle.java
+     *
+     * @param s The string to parse
+     * @return The parsed integer value
+     * @throws NumberFormatException if the number is invalid
+     */
+    public static int parseUid(String s) {
+        try {
+            return Integer.parseInt(s);
+        } catch (NumberFormatException e) {
```

Comment:
How often are the UIDs non-numeric?  It might be better from a performance perspective (and personal taste on my part) to not use NFE to control flow.  A simple regex like [1-9][0-9]* to ensure a number > 0 maybe?

Reply:
pretty often actually, its a good point, I'll update it with a regex.